### ISSUE TITLE
accessibility: add for atribut on ion-label and id on ion-input for correct display with VoiceOver

### ISSUE THREAD

**Issue by terrancle** [ionic-team/ionic#21298]
Hello,
To move from text field to text field in a form, VoiceOver has a function in the action rotor.
If I use a traditional ion-label and a traditional ion-input, there is no connection with rotor action navigation.
In html, so that VoiceOver can have the correct label for a text field, I will do:
<label for = "name"> Name </label>
<input id = "name" name = "name" type = "text">

How to do with ionic to add the for attribute to the ion-label tag, and an id corresponding to ion-input?
Thank you

**Comment by terrancle** [comment]
Example html code :
```
<label for="name">Your name</label>
<input id="name" name="name" type="text">``
```

**Comment by liamdebeasi** [comment]
Thanks for the issue. Ionic Framework should link the label and input automatically:

```
<ion-item>
  <ion-label>Message Label</ion-label>
  <ion-input placeholder="Type a message"></ion-input>
</ion-item>
```

Can you please provide steps to reproduce the issue you are experiencing?

**Comment by terrancle** [comment]
Precisely, if I do not want to add a placeholder, to avoid giving a double labeling of the field for users who do not move in the form with the actions rotor, this creates a disturbance for those who navigate with the rotor.
It should be understood that VoiceOver users can read a form in two different ways in the way of navigation.
The placeholder is good for those who move with the rotor, but it gives double labeling for the other way to navigate.
Conversely, if I don't put a placeholder, I disturb the users who navigate with the actions rotor.
The common solution for all navigation modes is the for attribute on ion-label, and the id attribute on ion-input, as recommended by aria.
I know it is not easy to understand. Do you think this optimization can be done?

**Comment by liamdebeasi** [comment]
Sorry, my code was confusing. The `placeholder` is not what links the two components. The linked label in this case is `Message Label`. Does the rotor not work when you do:

```
<ion-item>
  <ion-label>Message Label</ion-label>
  <ion-input></ion-input>
</ion-item>
```

**Comment by terrancle** [comment]
The rotor works, the text field label is not offered.
VoiceOver changes fields but does not state the ion-label.
The user therefore does not know what value to write.

**Comment by liamdebeasi** [comment]
As I mentioned previously, some steps to reproduce or a full code reproduction of the issue would be very helpful. I am not familiar with the VoiceOver rotor -- everything appears to be working fine for me.
